### PR TITLE
Credential Validation

### PR DESCRIPTION
**Pull request by franziskuskiefer**
Add credential validation step and some validation notes

### CODE REVIEW

**Review by keks — COMMENTED** [review]
I forgot pressing the "submit review" button :facepalm:

File: checksets/01-leaf_node.dhall [hunk 1]
```diff
@@ -36,7 +36,19 @@ let checks =
...
+                    (2) LeafNodeIn::into_verifiable_leaf_node
+                    ''
+                ,   ''
+                    (1) KeyPackageIn::validate
+                    ''
+                ,   ''
+                    (3) RatchetTree::try_from_nodes
+                    ''
+                ]
+            )
```
[the remark this answers is not in the document]
Comment (franziskuskiefer) [reply checksets/01-leaf_node.dhall:51]: I don't really understand how this is supposed to be used. I also didn't find documentation. If there's a logic to it, it needs to be documented somewhere.

**Comment by franziskuskiefer** [comment]
@keks what are the next steps here?

File: checksets/01-leaf_node.dhall (hunk 1 above)
[the remark this answers is not in the document]
Comment (franziskuskiefer) [reply checksets/01-leaf_node.dhall]: I moved it. Is this how it's supposed to work?

**Comment by keks** [comment]
I proposed some changes on this branch: https://github.com/openmls/validation.dhall/tree/keks/credential-validation-and-notes

**Comment by franziskuskiefer** [comment]
Looks good, let's get that in?

**Comment by franziskuskiefer** [comment]
@keks What's the state here?

**Comment by keks** [comment]
At least part of this doesn't belong into this check set, because it is about validation the app needs to do. We should discuss whether (an when) we want to launch a check set repository/registry for checks that need to be done by the application and move these there.

Otherwise I need to review again and see if anything in the PR belongs into this repo and get that merged

**Comment by keks** [comment]
This PR is only related to the validation that needs to be done by the application and doesn't belong here. closing.

**Pull request closed without merging**